source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,15 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern.substr(pattern_len - 2) == "/*") {
+    return strncmp(uriPattern.c_str(), uri, pattern_len - 1) == 0 && uri[pattern_len - 1];
+  }
+
+  return uriPattern == uri;
+}
+
 bool ContextImpl::dNSNameMatch(const std::string& dNSName, const char* pattern) {
```

Comment:
Thank you very much. Please also make sure the function signature (parameter naming&order) aligns with the uriMatch function :)